core/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -525,35 +525,42 @@ public void testParentObjectMapperAreNested() throws Exception {
     }
 
     public void testLimitNestedDocsDefaultSettings() throws Exception{
-        MapperService mapperService = createIndex("test1", Settings.builder().build()).mapperService();
+        Settings settings = Settings.builder().build();
+        MapperService mapperService = createIndex("test1", settings).mapperService();
         String mapping = XContentFactory.jsonBuilder().startObject().startObject("type").startObject("properties")
             .startObject("nested1").field("type", "nested").endObject()
             .endObject().endObject().endObject().string();
         DocumentMapper docMapper = mapperService.documentMapperParser().parse("type", new CompressedXContent(mapping));
+        long defaultMaxNoNestedDocs = MapperService.INDEX_MAPPING_NESTED_DOCS_LIMIT_SETTING.get(settings);
 
-        // parsing a doc with 3 nested objects succeeds
+        // parsing a doc with No. nested objects > defaultMaxNoNestedDocs fails
         XContentBuilder docBuilder = XContentFactory.jsonBuilder();
         docBuilder.startObject();
         {
             docBuilder.startArray("nested1");
             {
-                docBuilder.startObject().field("field1", "11").field("field2", "21").endObject();
-                docBuilder.startObject().field("field1", "12").field("field2", "22").endObject();
-                docBuilder.startObject().field("field1", "13").field("field2", "23").endObject();
+                for(int i=0; i<=defaultMaxNoNestedDocs; i++) {
+                    docBuilder.startObject().field("f", i).endObject();
```

Comment:
I'm suprised this triggers the no. fields warnings since "f" is not the field thats defined with the "nested" type above? I would have expected this to not trigger the limit but the exception tested below seems to indicate it triggers the setting. What am I missing?

Reply:
@cbuescher Please correct me if  I  misunderstood your previous comment: "could you change this to that the the document has more than the default number of nested docs (simple loop) and gets rejected". 
This is what this new test is about.
We have a parent nested field `nested1` under which we have many json objects with a field `f`, and since the number of these json objects is more than the default number, we get an error.

Reply:
Thanks, I probably read the test to quickly and missed the startArray() part. You are absolutely right, this is how I expected the test. Sorry for the noise.